SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -46,6 +47,27 @@ FSpatialLoadBalancingHandler::EvaluateActorResult FSpatialLoadBalancingHandler::
 	if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
 	{
 		AActor* NetOwner = GetReplicatedHierarchyRoot(Actor);
+
+		if (AController* Controller = Cast<AController>(Actor))
+		{
+			TArray<UActorComponent*> Components;
+			Controller->GetComponents(URemotePossessionComponent::StaticClass(), Components);
+			if (Components.Num() == 1)
+			{
+				// Only deal with the first one
+				if (URemotePossessionComponent* Component = Cast<URemotePossessionComponent>(Components[0]))
+				{
+					VirtualWorkerId TargetVirtualWorkerId;
+					if (Component->EvaluateMigration(NetDriver->LoadBalanceStrategy, TargetVirtualWorkerId))
```

Comment:
Can't we just return EvaluateSingleActor(Target)?

Reply:
What is the maintanence cost of this change? Do other engineers need to touch two places when modifying load balancing logic? If the only reason we can't use this is the if(!HasAuthority()) return.. could we not utilize a default lambda [Actor] { return !Actor->HasAuthority(); } and then for the nested case [Component] { return !Component->HasAuthority(); } ... if(Lambda())return....

Reply:
@yujunting ping

Reply:
It can't pass the check `if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))` too if we call the the EvaluateSingleActor without authority check.
If we want to make the function support actor and RemotePossessionComponent::Target, this function should be very complicated.